W-12142065: Metadata assignment operator (#211)
mulesoft/docs-dataweave: modules/ROOT/pages/dw-operators.adoc

  }
}
----

[[metadata-assignment-operator]]
== Metadata Assignment Operator

> DataWeave supports the metadata assignment operator `<~`, which enables you to set the metadata of any value without the need of using the `as` operator.

Reviewer: ```suggestion
DataWeave supports the metadata assignment operator `<~`, which enables you to set the metadata of any value without using the `as` operator.
```

> This is useful for cases where you want to set the metadata and keep the current value type. If coercing to a new type is also required then xref:dataweave-types-coercion.adoc[as] needs to be used.

Reviewer: ```suggestion
This operator is useful when you want to set the metadata and retain the current value type. If you are coercing the value to a new type, use xref:dataweave-types-coercion.adoc[as].
```

Reviewer: our doc style advises us that "this" identifies something, i.e., "this apple," "this operator," etc.

_Introduced in DataWeave 2.5.0. Supported by Mule 4.5 and later._

The operator syntax is as follows:

```
<valueB> <~ <objectA>
```

> and it reads `assigning objectA as metadata to valueB`. The arrow points to the value that has the metadata assigned.

Reviewer: ```suggestion
The operator assigns `objectA` as metadata to `valueB`. The arrow points to the value to which the metadata is assigned.
```

Versions prior to Mule 4.5 require the following syntax for instance to add the `class` metadata field to an object:

[source,weave,linenums]
----
%dw 2.0
---
{name: "Ken"} as Object {class: "Person"}
----

In Mule 4.5, the metadata assignment operator `<~` simplifies the syntax:

[source,weave,linenums]
----
%dw 2.0
---
{name: "Ken"} <~ {class: "Person"}
----

=== Metadata

> Metadata is defined as any additional information attached to a value that doesn't affect its identity. This means that given any two values

Reviewer: ```suggestion
Metadata is any additional information attached to a value that doesn't affect its identity. Given any two values
```

> `A` and `B` which are equal then if we attach metadata to `A` and we call it `A'` then `A == A' == B`. So we can define that a value is formed by the value itself

Reviewer: ```suggestion
`A` and `B` that are equal, if you attach metadata to `A` and call it `A'`, then `A == A' == B`. So you can define that a value that is formed by the value
```

Reviewer: we avoid using "we" in our docs per our doc style

we (haha) speak to the user as "you"
